src/state/src/State.c

```diff
@@ -54,6 +54,55 @@ STATUS createStateMachine(PStateMachineState pStates, UINT32 stateCount, UINT64
     return retStatus;
 }
 
+STATUS createStateMachineWithName(PStateMachineState pStates, UINT32 stateCount, UINT64 customData, GetCurrentTimeFunc getCurrentTimeFunc,
```

Comment:
In other implementation we have like this, basically the *only* difference between this and `createStateMachineWithName` and `createStateMachine` is that we are setting the name in the former.  I think it would be much cleaner here to call `createStateMachine` inside this function instead of re-doing (copy/paste) all that work here, and then just set the name as well, perhaps with the returned state machine object, or however is most convenient.